src/rust/ensogl/src/display/style.rs

```diff
@@ -0,0 +1,686 @@
+//! This module defines a cascading style sheet registry and related style management utilities.
+
+use crate::prelude::*;
+use crate::data::HashMapTree;
+use crate::data::Index;
+use crate::data::OptVec;
+
+
+
+// ============
+// === Data ===
+// ============
+
+/// Type of values in the style sheet.
+#[derive(Debug,Clone,PartialEq)]
+#[allow(missing_docs)]
+pub enum Data {
+    Invalid(String),
+    Number(f32)
+}
+
+/// Smart constructor for `Data`.
+pub fn data<T:Into<Data>>(t:T) -> Data {
+    t.into()
+}
+
+impl From<f32> for Data {
+    fn from(t:f32) -> Data {
+        Data::Number(t)
+    }
+}
+
+impl Mul<&Data> for &Data {
+    type Output = Data;
+    fn mul(self, rhs:&Data) -> Self::Output {
+        match(self,rhs) {
+            (Data::Invalid(t),_) => Data::Invalid(t.clone()),
+            (_,Data::Invalid(t)) => Data::Invalid(t.clone()),
+            (Data::Number(lhs),Data::Number(rhs)) => Data::Number(lhs*rhs),
+            // _ => Data::Invalid("Cannot multiply.".into())
+        }
+    }
+}
+
+impl Add<&Data> for &Data {
+    type Output = Data;
+    fn add(self, rhs:&Data) -> Self::Output {
+        match(self,rhs) {
+            (Data::Invalid(t),_) => Data::Invalid(t.clone()),
+            (_,Data::Invalid(t)) => Data::Invalid(t.clone()),
+            (Data::Number(lhs),Data::Number(rhs)) => Data::Number(lhs+rhs),
+            // _ => Data::Invalid("Cannot multiply.".into())
+        }
+    }
+}
+
+impl Eq for Data {}
+
+
+
+// ============
+// === Path ===
+// ============
+
+/// Path is a set of strings which describes how the variable or style sheet is nested in the
+/// cascading style sheet map.
+#[derive(Clone,Debug)]
+#[allow(missing_docs)]
+pub struct Path {
+    pub segments : Vec<String>
```

Comment:
As I read below that _the internal path representation is reversed_ I think this field should be named "reversed_segments"